src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -1921,7 +1820,91 @@ public static Vector128<float> Hypot(Vector128<float> x, Vector128<float> y)
             }
         }
 
-        /// <inheritdoc cref="ISimdVector{TSelf, T}.IsNaN(TSelf)" />
+        /// <inheritdoc cref="Vector64.IndexOf{T}(Vector64{T}, T)" />
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int IndexOf<T>(Vector128<T> vector, T value)
+        {
+            int result = BitOperations.TrailingZeroCount(Equals(vector, Create(value)).ExtractMostSignificantBits());
```

Comment:
The docs for this tzcnt intrinsic just say 'This method is similar in behavior to the x86 instruction TZCNT', and the docs for TZCNT say 'if source operand is zero, the content of destination operand are undefined'. Is this well-defined on all targets if none of the lanes match?

Reply:
`BitOperations` APIs in general are deterministic and ensure the handling of edge cases like `0` is well defined. In this case it will return `32`.

Notably the behavior of `TZCNT` is also not undefined for `0`, it is strictly defined to return the operand size. The consideration is that on hardware that does not report `TZCNT` support, the instruction is instead interpreted as `BSF` where the behavior is undefined; so you must ensure it is correctly guarded (via the relevant CPUID check) if you were to call such an intrinsic in C/C++ or directly in assembly (since it would not cause an illegal instruction fault).